src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -63,410 +64,517 @@
+    return object;
+
+}
+
+/**
+ * @brief Read macOS applications installed by homebrew.
+ *
+ * @param app_folder Path to the directory to scan.
+ * @param timestamp Timestamp of the scan to be attached to the events.
+ * @param random_id Id of the scan, to be attached to the events.
+ * @param queue_fd File descriptor of the queue to send events.
+ * @param LOCATION Identifier of the component to attach it to the events.
+ * @return 0 on success, 1 otherwise.
+ **/
+int sys_read_homebrew_apps(const char * app_folder, const char * timestamp, int random_id, int queue_fd, const char* LOCATION) {
+
+    DIR *dr;
+    char path[PATH_LENGTH];
+    struct dirent *de;
+    struct dirent *version = NULL;
+
+    // Define time to sleep between sent messages
+    int usec = 1000000 / wm_max_eps;
+
+
+    dr = opendir(app_folder);
+
+    if (dr == NULL) {
+        mtdebug1(WM_SYS_LOGTAG, "No homebrew applications found in '%s'", app_folder);
+        return 1;
+    }
+
+    while ((de = readdir(dr)) != NULL) {
+
+        if (strncmp(de->d_name, ".", 1) == 0) {
+            continue;
+        }
+
+        cJSON *object = cJSON_CreateObject();
+        cJSON *package = cJSON_CreateObject();
+        cJSON_AddStringToObject(object, "type", "program");
+        cJSON_AddNumberToObject(object, "ID", random_id);
+        cJSON_AddStringToObject(object, "timestamp", timestamp);
+        cJSON_AddItemToObject(object, "program", package);
+        cJSON_AddStringToObject(package, "format", "pkg");
+
+        // Some packages may come with a '@' followed up by the product's version.
+        // Since we already have said version, we can discard it.
+        char * tmp = NULL;
+        char * name_str = NULL;
+        os_strdup(de->d_name, name_str);
+        if (tmp = strchr(name_str, '@'), tmp != NULL) {
+            *tmp = '\0';
         }
-        cJSON_AddStringToObject(package, "location", app_folder);
-
-        if (invalid) {
-            char * program_name;
-            char * end;
-
-            // Extract program name from the path
-            program_name = strrchr(app_folder, '/');
-            program_name++;
-            end = strchr(program_name, '.');
-            *end = '\0';
-
-            if (normalize_mac_package_name(program_name, &vendor_name, &package_name)) {
-                cJSON_AddStringToObject(package, "name", package_name);
-                os_free(package_name);
-                if (vendor_name) {
-                    cJSON_AddStringToObject(package, "vendor", vendor_name);
-                    os_free(vendor_name);
+        cJSON_AddStringToObject(package, "name", name_str);
+        os_free(name_str);
+
+        snprintf(path, PATH_LENGTH - 1, "%s/%s", app_folder, de->d_name);
+        cJSON_AddStringToObject(package, "location", path);
+
+        DIR *dir;
+        dir = opendir(path);
+        if (dir != NULL) {
+            while ((version = readdir(dir)) != NULL) {
+                if (!strncmp(version->d_name, ".", 1)) {
```

Comment:
If I recall correctly,  the `..` dir entry had to be skipped too.

Reply:
Using `strncmp`, it should be caught by comparing the first char which is a dot in both cases